twill-core/src/main/java/org/apache/twill/internal/TwillContainerLauncher.java

```diff
@@ -225,5 +257,35 @@ public ContainerLiveNodeData getLiveNodeData() {
     public void kill() {
       processController.cancel();
     }
+
+    private void killAndWait(int maxWaitSecs) {
+      try {
+        Stopwatch watch = new Stopwatch();
+        watch.start();
+        int tries = 0;
+        while (watch.elapsedTime(TimeUnit.SECONDS) < maxWaitSecs) {
```

Comment:
Why we need to kill it in a loop? Is that mean the `processController.cancel()` doesn't guarantee it will kill the container?